### ISSUE TITLE
Proposal: Use template credentials fo `sam local`

### ISSUE THREAD

**Issue by rustyautopsy** [awslabs/serverless-application-model#865]
The goal here is to be able to properly test the configured IAM Roles/Policies for the Lambda(s) **before** deploying, running, dig through logs, find the error, update Lambda and repeat.

To do this I am thinking of doing the following, I would love to know if there is a better way.

- parse SAM template for Functions
- create an IAM user "sam-local-<function-name>"
- UPSERT ~/.aws/config & ~/.aws/credentials
- UPSERT Polices/Roles defined in the template to the "sam-local-<function-name>" user
- execute `sam local invoke… —profile sam-local-<function-name>`

**Comment by bramkoot** [comment]
I agree that this would be a great feature. In terms of implementation, I would much more prefer something like the policy simulator ([docs](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_testing-policies.html#policies-simulator-using-api)).
